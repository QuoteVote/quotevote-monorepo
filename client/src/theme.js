import { lightBlue, teal } from '@mui/material/colors'

// Named constants for fallback color values (official MUI values)
const LIGHT_BLUE_500_HEX = '#03a9f4'
const TEAL_A400_HEX = '#00bcd4'

// Provide safe fallbacks in case the color scales are undefined during
// early test-time resolution. Some tests import theme before modules are
// fully hoisted which can lead to lightBlue being undefined.
const safeLightBlue = (lightBlue && lightBlue[500]) || LIGHT_BLUE_500_HEX
const safeTealA400 = (teal && teal.A400) || TEAL_A400_HEX

const theme = {
  palette: {
    primary: {
      main: '#fff',
      contrastText: '#52b274',
    },
    secondary: {
      main: '#52b274',
      contrastText: '#fff',
    },
    background: {
      default: '#EEF4F9',
    },
  },
  activityCards: {
    quoted: {
      color: '#E36DFA',
      fontColor: '#000000',
    },
    commented: {
      color: '#FDD835',
      fontColor: '#000000',
    },
    upvote: {
      color: '#52b274',
      fontColor: '#000000',
    },
    downvote: {
      color: '#FF6060',
      fontColor: '#000000',
    },
    submitted: {
      color: '#000000',
      fontColor: '#000000',
    },
    hearted: {
      color: '#F16C99',
      fontColor: '#000000',
    },
    posted: {
      color: '#FFFFFF',
      fontColor: '#000000',
    },
    trending: {
      color: safeLightBlue,
      fontColor: '#000000',
    },
  },
  subHeader: {
    activeIcon: {
      color: safeTealA400,
    },
    default: {
      color: 'black',
    },
    followButton: {
      backgroundColor: '#52b274',
      color: 'white',
    },
  },
  activityCardsComplete: {
    peach: '#F44336',
    greenSecondary: '#4CAF50',
    // gray: rgba(25, 25, 25, 0.77),
    lightblueCard: '#00BCD4',
    orange: '#FF9801',
    gray1: '#454545',
    downvotedCardAndError: '#DA3849',
    blackCard: '#2D2A2A',
    greenPrimary: '#52b274',
    heartedPinkCard: '#F16C99',
    backgroundOffWhite: '#FAFAFA',
    mintyGreen: '#00E676',
    subsectionTitleMutedBlack: '#424556',
    blue: '#56B3FF',
    yellow: '#FEC02F',
    red: '#FF6060',
    violet: '#E36DFA',
    purple: '#791E89',
    gray2inactive: '#D8D8D8',
  },
  alerts: {
    info: '#00CAE3',
    success: '#55B559',
    warning: '#FF9E0F',
    danger: '#F55145',
    primary: '#A72ABD',
  },
}

export default theme
