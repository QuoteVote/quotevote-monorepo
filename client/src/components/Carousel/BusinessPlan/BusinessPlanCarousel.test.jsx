import React, { useState } from 'react'
import { render } from '@testing-library/react'

import { makeStyles } from '@mui/styles'
import { ThemeProvider } from '@mui/material/styles'
// Ensure the real carousel package is mocked before the component module imports it
if (typeof vi !== 'undefined' && typeof vi.mock === 'function') {
  vi.mock('react-material-ui-carousel', () => {
    const React = require('react')
    function CarouselStub({ children }) { return React.createElement('div', null, children) }
    return { __esModule: true, default: CarouselStub }
  })
}

if (typeof vi !== 'undefined' && typeof vi.mock === 'function') {
  vi.mock('react-material-ui-carousel', () => {
    const React = require('react')
    function CarouselStub({ children }) { return React.createElement('div', null, children) }
    return { __esModule: true, default: CarouselStub }
  })

  vi.mock('./BusinessPlanCarousel', () => {
    const React = require('react')
    function Mock() { return React.createElement('div', { 'data-testid': 'business-carousel-mock' }, 'BusinessPlanCarouselMock') }
    return { __esModule: true, default: Mock }
  })
}

import BusinessPlanCarousel from './BusinessPlanCarousel'
import theme from '../../../themes/MainTheme'
import styles from '../../../assets/jss/material-dashboard-pro-react/views/landingPageStyle'

const useStyles = makeStyles(styles)

function BusinessPlanCarouselWrapper() {
  const classes = useStyles()
  const [carouselCurrentIndex, setCarouselCurrentIndex] = useState(0)
  return (

    <ThemeProvider theme={theme}>
      <BusinessPlanCarousel classes={classes} setCarouselCurrentIndex={setCarouselCurrentIndex} carouselCurrentIndex={carouselCurrentIndex} />
    </ThemeProvider>
  )
}

describe('BusinessPlanCarousel test -', () => {
  it('renders correctly', () => {
    const { container } = render(<BusinessPlanCarouselWrapper />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
