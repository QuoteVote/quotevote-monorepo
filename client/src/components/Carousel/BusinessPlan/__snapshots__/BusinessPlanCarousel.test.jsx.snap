// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BusinessPlanCarousel test - > renders correctly 1`] = `
<div
  data-testid="business-carousel-mock"
>
  BusinessPlanCarouselMock
</div>
`;
